g3a_naturalmortality error when restricted age

I'm trying to have an addition natural mortality only to the first age (age0) of the immature stock (age0-5) as below

```
imm_actions <- 
  list(...

    ## natural mortality by age (age0-5)
    gadget3::g3a_naturalmortality(imm_stock, 
                              gadget3::g3a_naturalmortality_exp()),

    ## extra natural mortality only age0
    gadget3::g3a_naturalmortality(imm_stock,
                              gadget3::g3a_naturalmortality_exp(
                                   param_f = g3_parameterized('Mice', by_age = FALSE)),
                                   run_f = ~age == 0),
  ...)
```

the model returns a likelihood but fails to compile the object function.
```
> result <- model(tmb_param$value)
> print(result[[1]])
[1] 32.62687


>   obj.fun <- gadget3::g3_tmb_adfun(tmb_model, tmb_param)
Note: Using Makevars in /home/vano0001/.R/Makevars 
g++ -std=gnu++14 -I"/usr/share/R/include" -DNDEBUG -I"/home/vano0001/R/x86_64-pc-linux-gnu-library/4.2/TMB/include" -I"/home/vano0001/R/x86_64-pc-linux-gnu-library/4.2/RcppEigen/include"  -DTMB_SAFEBOUNDS -DTMB_EIGEN_DISABLE_WARNINGS -DLIB_UNLOAD=R_unload_g3_tmb_93317d4cd0b1e8e8cb381a15addb2c674558c454  -DTMB_LIB_INIT=R_init_g3_tmb_93317d4cd0b1e8e8cb381a15addb2c674558c454  -DCPPAD_FRAMEWORK     -fpic  -std=gnu++11 -Wno-ignored-attributes -DEIGEN_PERMANENTLY_DISABLE_STUPID_WARNINGS -O3 -flto -march=native -c /tmp/RtmpCjrwNy/g3_tmb_93317d4cd0b1e8e8cb381a15addb2c674558c454.cpp -o /tmp/RtmpCjrwNy/g3_tmb_93317d4cd0b1e8e8cb381a15addb2c674558c454.o
/tmp/RtmpCjrwNy/g3_tmb_93317d4cd0b1e8e8cb381a15addb2c674558c454.cpp: In member function ‘Type objective_function<Type>::operator()()’:
/tmp/RtmpCjrwNy/g3_tmb_93317d4cd0b1e8e8cb381a15addb2c674558c454.cpp:1122:14: error: ‘age’ was not declared in this scope
 1122 |         if ( age == 0 ) {
      |              ^~~
make: *** [/usr/lib/R/etc/Makeconf:177: /tmp/RtmpCjrwNy/g3_tmb_93317d4cd0b1e8e8cb381a15addb2c674558c454.o] Error 1
Error in TMB::compile(gsub("\\\\", "/", cpp_path), flags = paste(c("-std=gnu++11",  : 
  Compilation failed
```

I've tried with a different age or with a different syntax like `run_f = g3_formula(~age == 0)` but the issue persists.
Having `g3a_naturalmortality` expressed twice for a stock works, I've checked, also playing with `~cur_step == `. The issue seems linked to the `~age ==`

--- a/R/action_naturalmortality.R
+++ b/R/action_naturalmortality.R
@@ -19,9 +19,9 @@ g3a_naturalmortality <- function (
     stock__num <- g3_stock_instance(stock, 0)
     stock__wgt <- g3_stock_instance(stock, 1)
 
-    out[[step_id(run_at, stock, action_name)]] <- g3_step(f_substitute(~if (run_f) {
+    out[[step_id(run_at, stock, action_name)]] <- g3_step(f_substitute(~{
         debug_label("Natural mortality for ", stock)
-        stock_iterate(stock, {
+        stock_iterate(stock, if (run_f) {
             stock_ss(stock__num) <- stock_ss(stock__num) * (mortality_f)
         })
     }, list(

--- a/tests/test-action_naturalmortality.R
+++ b/tests/test-action_naturalmortality.R
@@ -26,6 +26,11 @@ actions <- list(
         prey_a,
         g3a_naturalmortality_exp(~naturalmortality_prey_a[[age - 3 + 1]]),
         run_f = ~cur_time > 0),  # NB: No mortality on the first step
+    g3a_naturalmortality(
+        prey_a,
+        # Spawning mortality, only for age 1, off by default
+        g3a_naturalmortality_exp(g3_parameterized('spawn_mortality', value = 0, optimise = TRUE)),
+        run_f = quote( age == 3 )),  # NB: Only for first age
     list(
         '999' = ~{
             if (cur_time == 0) {
@@ -105,3 +110,50 @@ ok_group("natural mortality", {
         gadget3:::ut_tmb_r_compare(model_fn, model_tmb, param_template)
     }
 })
+
+ok_group("Spawning natural mortality")
+params <- attr(model_fn, 'parameter_template')
+params['spawn_mortality'] <- 0.99
+
+result <- model_fn(params)
+r <- attributes(result)
+# str(result)
+# str(as.list(r), vec.len = 10000)
+
+# Step 0
+ok(ut_cmp_identical(as.vector(r$step0_prey_a__num), c(
+    30 * exp(-0.99 * 3 * (1/12)),
+    40,
+    50 )), "step0_prey_a__num: Natural mortality disabled by run_f, spawn natural mortality for first agegroup")
+ok(ut_cmp_identical(as.vector(r$step0_prey_a__wgt), c(300, 400, 500)), "step0_prey_a__wgt: Weight unchanged")
+
+# Step 1
+ok(ut_cmp_identical(as.vector(r$step1_prey_a__num), c(
+    30 * exp(-0.99 * 3 * (1/12))^2
+        * exp(-0.6 * 3 * (1/12)),
+    40 * exp(-0.7 * 3 * (1/12)),
+    50 * exp(-0.1 * 3 * (1/12)))), "step1_prey_a__num: Natural mortality reduced using exp(vec) & spawn natural mortality")
+ok(ut_cmp_identical(as.vector(r$step1_prey_a__wgt), c(300, 400, 500)), "step1_prey_a__wgt: Weight unchanged")
+
+# Step 2
+ok(ut_cmp_identical(as.vector(r$step2_prey_a__num), c(
+    30 * exp(-0.99 * 3 * (1/12))^2 * exp(-0.99 * 5 * (1/12))
+       * exp(-0.6 * 3 * (1/12)) * exp(-0.6 * 5 * (1/12)),
+    40 * exp(-0.7 * 3 * (1/12)) * exp(-0.7 * 5 * (1/12)),
+    50 * exp(-0.1 * 3 * (1/12)) * exp(-0.1 * 5 * (1/12)))), "step2_prey_a__num: Reduced again, used different step size")
+ok(ut_cmp_identical(as.vector(r$step2_prey_a__wgt), c(300, 400, 500)), "step2_prey_a__wgt: Weight unchanged")
+
+# Step 3
+ok(ut_cmp_identical(as.vector(r$step3_prey_a__num), c(
+    30 * exp(-0.99 * 3 * (1/12))^2 * exp(-0.99 * 5 * (1/12)) * exp(-0.99 * 1 * (1/12))
+       * exp(-0.6 * 3 * (1/12)) * exp(-0.6 * 5 * (1/12)) * exp(-0.6 * 1 * (1/12)),
+    40 * exp(-0.7 * 3 * (1/12)) * exp(-0.7 * 5 * (1/12)) * exp(-0.7 * 1 * (1/12)),
+    50 * exp(-0.1 * 3 * (1/12)) * exp(-0.1 * 5 * (1/12)) * exp(-0.1 * 1 * (1/12)))), "step3_prey_a__num: Reduced one more time, using another step size")
+ok(ut_cmp_identical(as.vector(r$step3_prey_a__wgt), c(300, 400, 500)), "step3_prey_a__wgt: Weight unchanged")
+
+if (nzchar(Sys.getenv('G3_TEST_TMB'))) {
+    param_template <- attr(model_cpp, "parameter_template")
+    param_template$value <- params[param_template$switch]
+    gadget3:::ut_tmb_r_compare(model_fn, model_tmb, param_template)
+}
+#### ok_group("Spawning natural mortality")

